pkg/common/scheduler/state/helpers.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package state
+
+import (
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/eventing-kafka/pkg/common/scheduler"
+)
+
+func PodNameFromOrdinal(name string, ordinal int32) string {
+	return name + "-" + strconv.Itoa(int(ordinal))
+}
+
+func OrdinalFromPodName(podName string) int32 {
+	ordinal, err := strconv.ParseInt(podName[strings.LastIndex(podName, "-")+1:], 10, 32)
+	if err != nil {
+		return math.MaxInt32
+	}
+	return int32(ordinal)
+}
+
+func Retry(f func() error) error {
```

Comment:
what about using this function instead: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Poll ?